src/components/avatar/Avatar.js

```diff
@@ -30,50 +30,38 @@ type Props = {
     size?: $Keys<typeof SIZES>,
 };
 
-type State = {
-    /** boolean to determine if image did not load correctly */
-    hasImageErrored: boolean,
-};
-
-class Avatar extends React.PureComponent<Props, State> {
-    state = {
-        hasImageErrored: false,
-    };
-
-    componentWillReceiveProps(nextProps: Props) {
-        if (this.state.hasImageErrored && this.props.avatarUrl !== nextProps.avatarUrl) {
-            this.setState({
-                hasImageErrored: false,
-            });
-        }
-    }
-
-    onImageError = () => {
-        this.setState({
-            hasImageErrored: true,
-        });
-    };
+function Avatar({ avatarUrl, className, name, id, size = '' }: Props) {
+    const [hasImageErrored, setHasImageErrored] = useState(false);
+    const classes = classNames(['avatar', className, { [`avatar--${size}`]: SIZES[size] }]);
 
-    render() {
-        const { avatarUrl, className, name, id, size = '' }: Props = this.props;
-        const { hasImageErrored }: State = this.state;
-        const classes = classNames(['avatar', className, { [`avatar--${size}`]: SIZES[size] }]);
-
-        let avatar;
-        if (avatarUrl && !hasImageErrored) {
-            avatar = <AvatarImage onError={this.onImageError} url={avatarUrl} />;
-        } else if (name) {
-            avatar = <AvatarInitials id={id} name={name} />;
-        } else {
-            avatar = <UnknownUserAvatar className="avatar-icon" />;
+    // Reset hasImageErrored state when avatarUrl changes
+    useEffect(() => {
+        if (hasImageErrored) {
```

Comment:
Is this check necessary or should we just reset `setHasImageErrored` to false whenever the url changes?

Reply:
good point. since this only fires when url changes, there should be no downside to just calling this always